Click tag on sentence page shows sentences w/same lang

alexandresmirnov opened the pull request:
This pull request addresses issue #829.

When clicking on a tag in the tag cloud to the right of a sentence (in the single sentence view), that tag now leads to a page with sentences in the same language as the original language.

Modified Tags helper, now has optional "$sentenceLang" parameter that
allows the tag link to point to generated page where all sentences are of
the same lang that the original sentence (where tag was clicked) was.

Modified sentences/show.ctp to pass $sentenceLang to
$tags->displayTagsModule

trang commented on app/views/helpers/tags.php:
```diff
@@ -102,14 +102,14 @@ public function displayTagsModule($tagsArray, $sentenceId = null)
      */
     public function displayTag(
-        $tagName, $tagId, $sentenceId, $userId, $username = null, $date = null
+        $tagName, $tagId, $sentenceId, $userId, $sentenceLang = null, $username = null, $date = null
```
When you modify the signature of a function to add a new parameter, it is generally better to add the new parameter at the very end, especially if it is an optional parameter. This way there is less risk to break something. If you choose to add it somewhere in the middle, you have to make sure to fix all the usages of the function.

In our case, you will need to fix the params in `app/views/tags/add_tag_post.ctp`.
Cf. https://github.com/Tatoeba/tatoeba2/search?utf8=%E2%9C%93&q=displayTag

trang commented:
Thanks for your PR @alexandresmirnov =)

Besides of my comment above, there's one more thing that would be nice: if you could do a `rebase -i` to squash your 2 last commits into one commit (https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i).

Also, if it happens to you again that right after you made a commit, you realize you forgot something in the last commit, you can use `git commit --amend` (https://www.atlassian.com/git/tutorials/rewriting-history/git-commit--amend).

alexandresmirnov commented:
Thank you for the suggestions!

Squashed the commits and changed the order of the parameters.

alexandresmirnov commented:
Thank you for your help!

Squashed the commits and changed the order of the parameters.

trang commented:
It seems like the commit with the indentation fix disappeared. Maybe you didn't push it?

alexandresmirnov commented:
No, I just rebased and squashed all of the commits (including the indentation issues) into one.  This commit contains all the changes I implemented, up to date :)

trang commented:
Oh I see, I thought that you were fixing the indentation for the whole file, in which case you should make it a separate commit. But you were just fixing it for your own changes, in which case it makes more sense indeed that you squash everything into one commit.
